src/Controller/CheckinRequestController.php

```diff
@@ -165,6 +214,8 @@ protected function updateCheckinRequest(CheckinRequest $checkinRequest, bool $su
         );
 
         // Finish job processing for non-cancellation requests.
+        // Note: Unclear why we don't just check for null
+        // $checkoutRequest->getCancelRequestId() here:
```

Comment:
Following our discussion, I think this is what is going on:

1. We were assuming that `cancelRequestId` being non-null would indicate a cancel request, but being non-null would indicate a checkin request. We are then setting a checkin id

2. We were assuming that a checkin request would come with a `jobId`, which would then be set to the `checkinJobId`. Semantically, we want to finish the job if it is a checkin request, which is why we are checking for a `checkinJobId`.